src/components/UsersDetailsCard.jsx

```diff
@@ -33,10 +33,45 @@ const tableHeaders = [
   },
 ];
 
+function reducer(state, action) {
+  return { ...state, ...action };
+}
+
 export default function UsersDetailsCard() {
   const { enqueueSnackbar } = useSnackbar();
 
-  const [selectedArray, setSelectedArray] = useState([]);
+  const [
+    { selectedArray, searchText, open, selectAll, buttonArray, selectedButton },
+    setState,
+  ] = useReducer(reducer, {
+    selectedArray: [],
+    searchText: "",
+    open: false,
+    selectAll: false,
+    buttonArray: [],
+    selectedButton: "bt_1",
+  });
+
+  function handleOpenClose() {
+    if (!open) setState({ open: true });
```

Comment:
Its better to remove extra if else statement.
try using this
```setState( { open: !open } )```

Reply:
removed the fucntion it self